src/main/java/core/selenium/WebDriverConfig.java

```
@@ -0,0 +1,82 @@
+package core.selenium;
+
+import core.selenium.webdrivers.BrowserType;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * WebDriverConfig, this class is in charge of the basic configuration of the browser.
+ *
+ * @author Maday Alcala
+ * @version 0.0.1
+ */
+public final class WebDriverConfig {
+    private static WebDriverConfig instance;
+    private static Properties properties;
+    private static InputStream inputProperties;
+    private static final String BROWSER = "browser";
+    private static final String IMPLICIT_WAIT_TIME = "implicitWaitTime";
+    private static final String EXPLICIT_WAIT_TIME = "explicitWaitTime";
+
+    /**
+     * Constructor of WebDriverConfig.
+     */
+    private WebDriverConfig() {
+        initialize();
+    }
+
+    /**
+     * This method verifies that there is only one instance of 'WebDriverConfig' and if there is not,
+     * it creates it, according to the Singleton pattern.
+     *
+     * @return an instance of WebDriverConfig.
+     */
+    public static WebDriverConfig getInstance() {
+        if (instance == null) {
+            instance = new WebDriverConfig();
+        }
+        return instance;
+    }
+
+    /**
+     * This method reads browser properties and initializes the basic browser characteristics.
+     */
+    public void initialize() {
+        try {
+            inputProperties = new FileInputStream("gradle.properties");
+            properties = new Properties();
+            properties.load(inputProperties);
+        } catch (Exception e) {
+            System.out.println("Something went wrong.");
```

If you just print a message, you will have problems later in the excecution and it will be difficutl to find the error. You should throw the exception or do something with that

---

Now I am using logs and a throw exception.